File: kotlin/lib/library/src/main/java/com/xplpc/message/Request.kt
package com.xplpc.message

import com.xplpc.core.XPLPC

class Request(functionName: String, vararg params: Param) {
    var data: String
        private set

    init {
        data = XPLPC.config.serializer.encodeRequest(functionName, *params)
    }
}

File: python/lib/src/xplpc/message/request.py
from xplpc.core.xplpc import XPLPC
from xplpc.message.param import Param


class Request:
    def __init__(self, function_name: str, params: list[Param] | None = None):
        self.function_name = function_name
        self.params = params

    def data(self) -> str:
        return XPLPC().config.serializer.encode_request(
            self.function_name,
            self.params,
        )
